app/lib/steps/logic.rb

```diff
@@ -12,10 +12,14 @@ def upper_tribunal?(session_data)
         if controlled?(session_data)
           session_data["immigration_or_asylum"]
         else
-          session_data["matter_type"].in?(%w[immigration asylum])
+          session_data["immigration_or_asylum_type_upper_tribunal"].in?(%w[immigration_upper asylum_upper]) && !domestic_abuse_applicant?(session_data)
         end
       end
 
+      def domestic_abuse_applicant?(session_data)
+        session_data["domestic_abuse_applicant"]
```

Comment:
If we're being extra thorough, this should really have a `!controlled?(session_data)` as well - just in case someone changes their answers after saying yes to this question